### PR TITLE
Refactor `Config` module to use environment variables, improve validation (DBAI-18)

### PR DESCRIPTION
**Pull request by ssciolla**
This PR aims to resolve DBAI-18. It is still a draft.

To Do
- [x] Replace `config.yml` with `.env`
- [x] Change `ConfigModule` to look for environment variable keys
- [x] Extract `CheckableData` and `Check` classes.
- [x] Update usages of `ConfigService`
- [x] Add unit tests (I did it backwards ☹️ )
- [x] Support `.env` outside of Docker (https://github.com/bkeepers/dotenv)
- [x] Update documentation
- [x] Change Docker database settings names 

Resource(s)
- https://12factor.net/config
- https://github.com/bkeepers/dotenv

### CODE REVIEW

**Review by jayamala17 — APPROVED** [review]

File: lib/config.rb [hunk 1]
```diff
@@ -110,142 +208,109 @@ module Config
...
-                  location_uuid: verify_string("LocationUUID", api_data["LocationUUID"])
-                ),
-                remote: create_remote_config(remote_data)
-              )
-            end
-          )
+          name: data.get_value(key: "REPOSITORY_NAME"),
+          description: data.get_value(key: "REPOSITORY_DESCRIPTION")
         ),
+        database: db_data.keys.length > 0 ? create_database_config(db_data) : nil,
```
[the remark this answers is not in the document]
Comment (jayamala17) [reply lib/config.rb:294]: db_data.keys.length > 0 requires parentheses

File: lib/config.rb [hunk 2]
```diff
@@ -1,8 +1,106 @@
...
+    end
+
+    def reason
+      raise NotImplementedError
+    end
+  end
+
+  class NotNilCheck < CheckBase
+    def check?(value)
+     !value.nil?
```
Comment (jayamala17) [inline lib/config.rb:20]: Indentation error

File: lib/config.rb [hunk 3]
```diff
@@ -110,142 +208,109 @@ module Config
     keyword_init: true
   )
 
-  class ConfigError < StandardError
-  end
-
   class ConfigService
-    include SemanticLogger::Loggable
+    ARCHIVEMATICA_INSTANCES = ["ARCHIVEMATICA_DEV",  "ARCHIVEMATICA_LAB", "ARCHIVEMATICA_AMI", "ARCHIVEMATICA_VGA"]
```
[the remark this answers is not in the document]
Comment (ssciolla) [reply lib/config.rb]: Fixed. https://github.com/mlibrary/bag-courier/pull/38/commits/6f62a2643de7761cf4c3481c1fbfe8abd9801aaf

File: lib/config.rb (hunk 2 above)
Comment (ssciolla) [reply lib/config.rb]: Fixed. https://github.com/mlibrary/bag-courier/pull/38/commits/6f62a2643de7761cf4c3481c1fbfe8abd9801aaf

File: lib/config.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ssciolla) [reply lib/config.rb]: Fixed. https://github.com/mlibrary/bag-courier/pull/38/commits/6f62a2643de7761cf4c3481c1fbfe8abd9801aaf

File: lib/config.rb [hunk 4]
```diff
@@ -110,142 +208,109 @@ module Config
...
+      data = CheckableData.new(data)
+      db_data = data.get_subset_by_key_stem("DATABASE_")
+
+      arch_configs = []
+      ARCHIVEMATICA_INSTANCES.each do |instance_name|
+        instance_data = data.get_subset_by_key_stem(instance_name + "_")
+        arch_configs << create_archivematica_config(instance_data) if instance_data.keys.length > 0
+      end
+
+      config = Config.new(
```
Comment (jayamala17) [inline lib/config.rb]: config value assigned but not used.

**Review by jayamala17 — APPROVED** [review]

**Pull request merged**